File: Contests/1647/A.cpp
#include <bits/stdc++.h>
using namespace std;

bool checkZero(string num) {
    for (int i = 0 ; i < num.length() ; i++) {
        if(num[i] == '0')
            return 0;
    } 
}

int sumOfDigits (int num) {
    string strNum = to_string(num);
    if (!(checkZero(strNum)))
        return 0;
    int sum = 0;
    for (int i = 0 ; i < strNum.length() ; i++) {
        sum += (int)strNum[i] - 48;
    }
    return sum;
}

int number (int n) {
    int i = 1;
    while( n >= sumOfDigits(i)) {
        i++;
        if (!(sumOfDigits(i)))
            continue;
    }
    return i;
}

int main() {
    cout << number(3);
}

File: Contests/1647/A.py
def sumOfDigits (strN):
    sum = 0
    for i in strN:
        sum += int(i)

    return sum

def checkZero(strN):
    for i in strN:
        if i == '0':
            return True
    return False

def checkNumber(strN, n):
    if ((str(n) in strN) and (checkZero == True)):
        return False
    return True

def numIncrement(n):
    i = 0
    num = []
    while (checkNumber(str(i), n) == True):
        if sumOfDigits(str(i)) == n:
            num.append(i)
            i+=1
            if ( i == n ):
                i+=1
        else:
            i+=1
            if ( i == n ):
                i+=1
            continue

    return i

print(numIncrement(3))